db.py

```diff
@@ -1,13 +1,151 @@
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    course_id = db.Column(db.Integer)
+    course_year = db.Column(db.Integer)
+
+    # Creation of a link to the compound key (id, year) of the course table
+    __table_args__ = (
+        db.ForeignKeyConstraint(
+            ['course_id', 'course_year'],
+            ['course.id', 'course.year']
+        ),
+    )
+
+    user = db.relationship('User', backref=db.backref('user_teacher', uselist=False))
+    course = db.relationship('Course', backref=db.backref('course_teacher', lazy=True))
+
+
+class PreferenceAssignment(db.Model):
+    __tablename__ = 'preference_assignment'
+    id = db.Column(db.Integer, primary_key=True)
+    course_id = db.Column(db.Integer, nullable=False)
+    course_year = db.Column(db.Integer, nullable=False)
+    researcher_id = db.Column(db.Integer, db.ForeignKey('researcher.id'), nullable=False)
+
+    # Creation of a link to the compound key (id, year) of the course table
+    __table_args__ = (
+        db.UniqueConstraint('course_id', 'course_year', 'researcher_id', name='unique_course_year_researcher'),
+        db.ForeignKeyConstraint(
+            ['course_id', 'course_year'],
+            ['course.id', 'course.year']
+        ),
+    )
+
+    course = db.relationship('Course', backref=db.backref('course_preference_assignment', lazy=True))
+    researcher = db.relationship('Researcher', backref=db.backref('researcher_preference_assignment', lazy=True))
+
+
+class Configuration(db.Model):
+    __tablename__ = 'configuration'
+    id = db.Column(db.Integer, primary_key=True)
+    year = db.Column(db.Integer, nullable=False)
+    is_current_year = db.Column(db.Boolean, default=True)
+
+    @classmethod
+    def update_current_year(cls, config_id):
+        # Update all years to set is_current_year to False
+        cls.query.update({cls.is_current_year: False})
+
+        selected_config = cls.query.get(config_id)
+        if selected_config:
+            selected_config.is_current_year = True
+            db.session.commit()
+
+
+class Organization(db.Model):
+    __tablename__ = 'organization'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String, nullable=False, unique=True)
+
+    users = db.relationship('User', back_populates='organization')
+    courses = db.relationship('Course',
+                              secondary='course_organization',
+                              back_populates='organizations',
+                              primaryjoin="Organization.id == CourseOrganization.organization_id",
+                              secondaryjoin="and_(Course.id == CourseOrganization.course_id, Course.year == CourseOrganization.course_year)")
+
+
+class CourseOrganization(db.Model):
+    __tablename__ = 'course_organization'
+
+    course_id = db.Column(db.Integer, db.ForeignKey('course.id'), primary_key=True)
+    course_year = db.Column(db.Integer, db.ForeignKey('course.year'), primary_key=True)
```

Comment:
```suggestion
    course_year = db.Column(db.Integer, primary_key=True)
```

The ForeignKeyConstraint is specified below